addons/binding/org.openhab.binding.neeo/src/main/java/org/openhab/binding/neeo/handler/NeeoDeviceHandler.java

```diff
@@ -0,0 +1,304 @@
+import org.openhab.binding.neeo.internal.NeeoBrainApi;
+import org.openhab.binding.neeo.internal.NeeoDeviceConfig;
+import org.openhab.binding.neeo.internal.NeeoDeviceProtocol;
+import org.openhab.binding.neeo.internal.NeeoHandlerCallback;
+import org.openhab.binding.neeo.internal.NeeoRoomConfig;
+import org.openhab.binding.neeo.internal.StatefulHandlerCallback;
+import org.openhab.binding.neeo.internal.models.NeeoDevice;
+import org.openhab.binding.neeo.internal.models.NeeoDeviceDetails;
+import org.openhab.binding.neeo.internal.models.NeeoDeviceDetailsTiming;
+import org.openhab.binding.neeo.internal.models.NeeoMacro;
+import org.openhab.binding.neeo.internal.models.NeeoRoom;
+import org.openhab.binding.neeo.internal.type.UidUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An extension of {@link AbstractThingHandler} that is responsible for handling commands for a room
+ *
+ * @author Tim Roberts - Initial contribution
+ */
+public class NeeoDeviceHandler extends AbstractThingHandler {
+
+    /** The logger */
+    private final Logger logger = LoggerFactory.getLogger(NeeoDeviceHandler.class);
+
+    /**
+     * The initialization task (null until set by {@link #initializeTask()} and set back to null in {@link #dispose()}
+     */
+    private final AtomicReference<Future<?>> initializationTask = new AtomicReference<>(null);
+
+    /**
+     * The refresh task (null until set by {@link #initializeTask()} and set back to null in {@link #dispose()}
+     */
+    private final AtomicReference<ScheduledFuture<?>> refreshTask = new AtomicReference<>(null);
+
+    /** The {@link NeeoDeviceProtocol} (null until set by {@link #initializationTask}) */
+    private final AtomicReference<NeeoDeviceProtocol> deviceProtocol = new AtomicReference<>();
+
+    /**
+     * Instantiates a new neeo room handler.
+     *
+     * @param typeGenerator the non-null type generator
+     */
+    NeeoDeviceHandler(Thing thing) {
+        super(thing);
+        Objects.requireNonNull(thing, "thing cannot be null");
+    }
+
+    /**
+     * Handles commands sent to the room
+     *
+     * @see
+     *      org.eclipse.smarthome.core.thing.binding.ThingHandler#handleCommand(org.eclipse.smarthome.core.thing.ChannelUID,
+     *      org.eclipse.smarthome.core.types.Command)
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        Objects.requireNonNull(channelUID, "channelUID cannot be null");
+        Objects.requireNonNull(command, "command cannot be null");
+
+        final NeeoDeviceProtocol protocol = deviceProtocol.get();
+        if (protocol == null) {
+            logger.debug("Protocol is null - ignoring update: {}", channelUID);
+            return;
+        }
+
+        final String groupId = channelUID.getGroupId();
+        final String channelId = channelUID.getIdWithoutGroup();
+
+        ((StatefulHandlerCallback) protocol.getCallback()).removeState(channelUID.getId());
```

Comment:
Why are you caching the old state? You say that you only want to communicate if it has changed, but why is that? is the communication with the device really that expensive in terms of it takes **very** long or consumes a lot of computation power on the brain?